### ISSUE TITLE
Multiple DynamoStream triggers on a function

### ISSUE THREAD

**Issue by pepso** [aws-amplify/amplify-cli#6748]
**Describe the bug**
When you have existing function with dynamo stream trigger, and you add second dynamodb trigger, when deploying to cloud, your lambda loses access to the previous trigger. 

**Amplify CLI Version**
4.43 (This bug have been around for long... 4.29.X onward I think)

**To Reproduce**
1. Create lambda function (triggerMe)
2. Create dynamo table table1
2. Create dynamo stream trigger for function (triggerMe)
3. Create dynamo table table2
4. Create dynamo stream trigger for function (triggerMe)
5. Deploy to cloud

**Expected behavior**
Both tables trigger dynamo stream and the function

**Current behavior**
Only one table trigger the strem processing correctly.

triggerMe function's cloud-formation template has 2 triggers defined with same "PolicyName": "lambda-trigger-policy". When this is deployed to cloud, latter one in the process overwrites the policy of the earlier. Basically one of the dynamoDB table's streams tab show the function "failed" and the reason being the lambda function doesn't have access to the stream.

e.g. triggerMe's cloud-formation template contains:
table1TriggerPolicy <- here the actual policy-name is "lambda-trigger-policy"
table1Trigger
table2TriggerPolicy <- here the actual policy-name is also "lambda-trigger-policy"
table2Trigger

**Workaround**
Manually update PolicyName in the triggered function's cloud-formation template to be unique per trigger source.

Basically rename triggerMe's cloud-formation template:
table2TriggerPolicy <- policy-name "lambda-trigger-policy" -> policy-name "lambda-trigger-policy-table2"
And this avoids the cloud deployment overwrite issue in IAM.

Let me know if you need more details, but this should be easy to replicate. I have had this issue since the dynamoDB triggers (~5 months ago), but haven't just managed to write this report before...

**Comment by akshbhu** [comment]
Thnaks for reporting it and the workaround too 👍

**Comment by akshbhu** [comment]
closing in favor of #6980
